### PR TITLE
Initial Geico meta layer

### PR DESCRIPTION
**Pull request by khemkav**
Added meta-geico to support Geico servers

### CODE REVIEW

**Review by c3-geico — DISMISSED** [review]
+1

File: meta-geico/recipes-kernel/linux/linux-aspeed/quartz.dts
Comment (Kwongyhue) [inline meta-geico/recipes-kernel/linux/linux-aspeed/quartz.dts:1]: Do we have permissions to make this public? This is the public Geico repo

File: meta-geico/conf/templates/default/bblayers.conf.sample [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+LCONF_VERSION = "8"
+
+BBPATH = "${TOPDIR}"
+BBFILES ?= ""
+
+BBLAYERS ?= " \
+  ##OEROOT##/meta \
+  ##OEROOT##/meta-openembedded/meta-oe \
+  ##OEROOT##/meta-openembedded/meta-networking \
+  ##OEROOT##/meta-openembedded/meta-python \
```
Comment (Kwongyhue) [inline meta-geico/conf/templates/default/bblayers.conf.sample:12]: I don't recall if this is explicitly required. If there's dependencies for this, it's fine. If not, would remove it

Comment (khemkav) [reply meta-geico/conf/templates/default/bblayers.conf.sample:12]: These are required for native tools to build

File: meta-geico/recipes-kernel/linux/linux-aspeed/quartz.dts
Comment (khemkav) [reply meta-geico/recipes-kernel/linux/linux-aspeed/quartz.dts:1]: This is public Geico but not outside the Geico. So only accessible to Geico associate

Comment (richardgeico) [reply meta-geico/recipes-kernel/linux/linux-aspeed/quartz.dts:1]: This is public and is accessible to the world

Comment (khemkav) [reply meta-geico/recipes-kernel/linux/linux-aspeed/quartz.dts:1]: Yes, me and Rama just checked, it is publically available to everyone

Comment (khemkav) [reply meta-geico/recipes-kernel/linux/linux-aspeed/quartz.dts:1]: This is GPL2 so we are supposed to opensource for any change in kernel

File: meta-geico/conf/templates/default/bblayers.conf.sample (hunk 1 above)
Comment (Kwongyhue) [reply meta-geico/conf/templates/default/bblayers.conf.sample:12]: Meant the python meta layer specifically, but yeah that should be fine.

**Review by Kwongyhue — DISMISSED** [review]

**Review by tvanfleet — DISMISSED** [review]

File: meta-geico/conf/machine/quartz.conf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+# Disable hardware root of trust by default.
+SOCSEC_SIGN_ENABLE = "0"
+
+# Flash size is 64M
+FLASH_SIZE = "65536"
+
+require conf/machine/include/ast2600.inc
+require conf/machine/include/obmc-bsp-common.inc
+
+SERIAL_CONSOLES = "115200;ttyS4"
```
Comment (PinkMonstera) [inline meta-geico/conf/machine/quartz.conf:16]: Serial console should be an interpolated value

File: meta-geico/conf/templates/default/local.conf.sample [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+#SSTATE_MIRRORS ?= "\
+#file://.* http://someserver.tld/share/sstate/PATH;downloadfilename=PATH \n \
+#file://.* file:///some/local/dir/sstate/PATH"
+
+#
+# Yocto Project SState Mirror
+#
+# The Yocto Project has prebuilt artefacts available for its releases, you can enable
+# use of these by uncommenting the following lines. This will mean the build uses
+# the network to check for artefacts at the start of builds, which does slow it down
```
Comment (PinkMonstera) [inline meta-geico/conf/templates/default/local.conf.sample:207]: artefacts - "artifacts"

File: meta-geico/conf/templates/default/local.conf.sample [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+# You can also enable the Gtk UI frontend, which takes somewhat longer to build, but adds
+# a handy set of menus for controlling the emulator.
+#PACKAGECONFIG:append:pn-qemu-system-native = " gtk+"
+
+#
+# Hash Equivalence
+#
+# Enable support for automatically running a local hash equivalence server and
+# instruct bitbake to use a hash equivalence aware signature generator. Hash
+# equivalence improves reuse of sstate by detecting when a given sstate
```
Comment (PinkMonstera) [inline meta-geico/conf/templates/default/local.conf.sample:235]: sstate - "state"

File: meta-geico/conf/machine/quartz.conf (hunk 2 above)
Comment (khemkav) [reply meta-geico/conf/machine/quartz.conf:16]: This is yocto variables and format

File: meta-geico/conf/templates/default/local.conf.sample (hunk 4 above)
Comment (khemkav) [reply meta-geico/conf/templates/default/local.conf.sample:235]: this is sstate only as it is part of sstate variable and not state. If you follow above section it is all about sstate

File: meta-geico/conf/templates/default/local.conf.sample (hunk 3 above)
Comment (khemkav) [reply meta-geico/conf/templates/default/local.conf.sample:207]: fixed

**Review by tvanfleet — APPROVED** [review]

**Review by Kwongyhue — APPROVED** [review]

**Review by c3-geico — APPROVED** [review]
+1

**Pull request merged**